ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/KafkaConfigStore.java

```diff
@@ -0,0 +1,238 @@
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(ksqlProperties);
+    }
+  }
+
+  private static class KafkaWriteOnceStore<V> {
+    private final String topicName;
+    private final String key;
+    private final Deserializer<V> deserializer;
+    private final Supplier<KafkaConsumer<String, byte[]>> consumerSupplier;
+    private final Supplier<KafkaProducer<String, V>> producerSupplier;
+
+    KafkaWriteOnceStore(
+        final String topicName,
+        final String key,
+        final Deserializer<V> deserializer,
+        final Supplier<KafkaConsumer<String, byte[]>> consumerSupplier,
+        final Supplier<KafkaProducer<String, V>> producerSupplier) {
+      this.topicName = topicName;
+      this.key = key;
+      this.deserializer = deserializer;
+      this.consumerSupplier = consumerSupplier;
+      this.producerSupplier = producerSupplier;
+    }
+
+    private Optional<V> read() {
+      final TopicPartition topicPartition = new TopicPartition(topicName, 0);
+      final List<TopicPartition> topicPartitionAsList = Collections.singletonList(topicPartition);
+
+      try (KafkaConsumer<String, byte[]> consumer = consumerSupplier.get()) {
+        consumer.assign(topicPartitionAsList);
+        consumer.seekToBeginning(topicPartitionAsList);
+
+        final Map<TopicPartition, Long> offsets = consumer.endOffsets(topicPartitionAsList);
+        final long endOffset = offsets.get(topicPartition);
+        while (consumer.position(topicPartition) < endOffset) {
+          log.debug(
+              "Reading from topic %s. Position(%d) End(%d)",
+              topicName,
+              consumer.position(topicPartition),
+              endOffset);
+          final ConsumerRecords<String, byte[]> records
+              = consumer.poll(Duration.of(5, ChronoUnit.SECONDS));
+          final Optional<ConsumerRecord<String, byte[]>> record =
+              records.records(topicPartition)
+                  .stream()
+                  .filter(r -> r.key().equals(key))
+                  .findFirst();
+          if (!record.isPresent()) {
+            continue;
+          }
+          final byte[] value = record.get().value();
+          log.debug("Found existing value in topic %s", topicName);
+          return Optional.of(deserializer.deserialize(topicName, value));
+        }
+        log.debug("No value found on topic %s", topicName);
+        return Optional.empty();
+      }
+    }
+
+    private void write(final String topicName, final V value) {
+      try (KafkaProducer<String, V> producer = producerSupplier.get()) {
+        producer.send(new ProducerRecord<>(topicName, key, value));
+        producer.flush();
+      }
+    }
+
+    V readMaybeWrite(final V value) {
+      final Optional<V> properties = read();
+      if (properties.isPresent()) {
+        return properties.get();
+      }
+
+      log.info("Writing current config to config topic");
```

Comment:
Still think this is spam and should be written at debug level.

Maybe you could log at info if you included some information about _what_ it being written, e.g. key & value.  Without them it's just spam IMHO.